Bottom: 93dbeaee8bdcfe763575e275bfbb0a6b803369a2
Top:    4e2a5fee0377c5322721a50b8c8eb148d6cfb9b0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-01-12 01:07:35 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5cd245f..c5cab7d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
